src/TNodeImp.h

```diff
@@ -0,0 +1,80 @@
+//
+// Created by shvmo on 29/04/2020.
+//
+
+#ifndef DATASTRUCTURESWET1_TNODEIMP_H
+#define DATASTRUCTURESWET1_TNODEIMP_H
+
+
+#include "TNode.h"
```

Comment:
please remove this include ty

Reply:
Why? Ifdef knows how to deal with cyclic includes

Reply:
i know.
it will work. 
but why should it be there?